modules/jdbc-test/src/test/java/org/testcontainers/jdbc/JDBCDriverTest.java

```diff
@@ -86,98 +85,91 @@ public static void testCleanup() {
 
     @Test
     public void test() throws SQLException {
-        performSimpleTest(jdbcUrl);
+        try (HikariDataSource dataSource = getDataSource(jdbcUrl, 1)) {
+            performSimpleTest(dataSource);
 
-        if (options.contains(Options.ScriptedSchema)) {
-            performTestForScriptedSchema(jdbcUrl);
-        }
+            if (options.contains(Options.ScriptedSchema)) {
+                performTestForScriptedSchema(dataSource);
+            }
 
-        if (options.contains(Options.JDBCParams)) {
-            performTestForJDBCParamUsage(jdbcUrl);
-        }
+            if (options.contains(Options.JDBCParams)) {
+                performTestForJDBCParamUsage(dataSource);
+            }
 
-        if (options.contains(Options.CharacterSet)) {
-            //Called twice to ensure that the query string parameters are used when
-            //connections are created from cached containers.
-            performSimpleTestWithCharacterSet(jdbcUrl);
-            performSimpleTestWithCharacterSet(jdbcUrl);
+            if (options.contains(Options.CharacterSet)) {
+                performSimpleTestWithCharacterSet(jdbcUrl);
 
-            performTestForCharacterEncodingForInitialScriptConnection(jdbcUrl);
-        }
+                performTestForCharacterEncodingForInitialScriptConnection(dataSource);
+            }
 
-        if (options.contains(Options.CustomIniFile)) {
-            performTestForCustomIniFile(jdbcUrl);
+            if (options.contains(Options.CustomIniFile)) {
+                performTestForCustomIniFile(dataSource);
+            }
         }
     }
 
-    private void performSimpleTest(String jdbcUrl) throws SQLException {
-        try (HikariDataSource dataSource = getDataSource(jdbcUrl, 1)) {
-            boolean result = new QueryRunner(dataSource, options.contains(Options.PmdKnownBroken)).query("SELECT 1", rs -> {
-                rs.next();
-                int resultSetInt = rs.getInt(1);
-                assertEquals("A basic SELECT query succeeds", 1, resultSetInt);
-                return true;
-            });
-
-            assertTrue("The database returned a record as expected", result);
-        }
-    }
+    private void performSimpleTest(HikariDataSource dataSource) throws SQLException {
+        boolean result = new QueryRunner(dataSource, options.contains(Options.PmdKnownBroken)).query("SELECT 1", rs -> {
+            rs.next();
+            int resultSetInt = rs.getInt(1);
+            assertEquals("A basic SELECT query succeeds", 1, resultSetInt);
```

Comment:
I'm probably missing something (and it's just copied from the original code), but why don't we `return resultSetInt == 1` instead of the `assertEquals`?

Reply:
this is just an indentation change. Let's discuss/change it separately :)